daemon/api_sideload_n_try.go

```diff
@@ -32,103 +34,99 @@ import (
 	"github.com/snapcore/snapd/client"
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/overlord/assertstate"
-	"github.com/snapcore/snapd/overlord/auth"
 	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/snap/snapfile"
 )
 
-const maxReadBuflen = 1024 * 1024
+type Form struct {
+	Values   map[string][]string
+	FileRefs map[string][]*FileReference
+}
+
+type FileReference struct {
+	Filename string
+	TmpPath  string
+}
 
-func sideloadOrTrySnap(c *Command, body io.ReadCloser, boundary string, user *auth.UserState) Response {
+func (f *Form) RemoveAll() {
+	for _, refs := range f.FileRefs {
+		for _, ref := range refs {
+			if err := os.Remove(ref.TmpPath); err != nil {
+				logger.Noticef("cannot remove temporary file: %v", err)
+			}
+		}
+	}
+}
+
+// SnapFileNameAndPath returns the original file path/name and the path to
+// where the temp file is written.
+func (f *Form) SnapFileNameAndPath() (name, path string, apiErr *apiError) {
```

Comment:
Nitpick, maybe `SnapFileNameAndTempPath() (name, tempath string, apiErr *apiError) {` for clarity?